Form UI/UX Enhancement

Soare-Robert-Daniel opened the pull request:
<!-- Issues that this pull request closes. -->
Closes #1330.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

### Summary
<!-- Please describe the changes you made. -->

New UI and UX. Better integration with Global Defaults.

Explication for the codes that appears on the front for clients:

- 101: The request is missing data like Form Id/Option, page URL, or nonce value - this can be missing when the Nonce Field in the form was manually deleted with force. Also, the given nonce might be invalid.
- 102: No given captcha token when Bot Protection is set to active.
- 104: The email field is missing. This is requested when the Form is set to subscribe the user to a mail list
- 106: The email could not be sent. Might be a problem with the mail service.
- 108: The API key for the Marketing Integration service is missing. Either it was deleted by hand or a script.
- 201: An external script has changed the settings, and the accessed service is no longer present.
- 202: Error from the service when trying to subscribe the user, like servers are unavailable, rejected request, etc.
- 203: The 3rd-party authentication is invalid. Might be a problem with API Key: expired, rejected, etc. Need to check with the provider.
- 204: The provided captcha token is invalid: the user might be a bot.
- 205: The API Key format is invalid. The client might not have copied the right key, or the provider changed the format, and we did not update the check.
- 206: The email was already registered in the Marketing Integration service.
- 207: The service rejected the given email by the user. It might consider it fake.

### Screenshots <!-- if applicable -->

----

### Test instructions
<!-- Describe how this pull request can be tested. -->

This is a UI refresh; we need to ensure that all style options are usable.

1. Make sure that the old blocks don't break.
2. Test the Sidebar and the responsive options.
3. Test the Global defaults

With this update, we let the user set a message in case of an error. The reason for the error will be displayed in the Console along with its code. Like this:

![image](https://user-images.githubusercontent.com/17597852/210768058-e7496734-94ff-4fe1-833e-8ddfefb687ed.png)

The basic email functionality should be effaced by this. If you happen to have a problem, please put an SS with an error in the console. The error code is vital for debugging.

<!--
#### Query
```javascript
new QueryQA().select('blocks').run()
```
-->

---- 

### Checklist before the final review

- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.
- [ ] Copy/Paste is working if the attributes are modified.

Soare-Robert-Daniel commented:
@HardeepAsrani @mghenciu I ported the tab navigation from Tabs. As discussed, the non-input field will be ignored. Is the behaviour from the video ok?

https://user-images.githubusercontent.com/17597852/207616335-6636046a-ea0a-4a6f-b50a-d6d44f29888f.mp4

HardeepAsrani commented:
@Soare-Robert-Daniel Looks good, can you also put "Textarea" in field type Option control? If the user selects that, we simply migrate from one block to anther.

@ineagu Let me know what you think about this.

mghenciu commented:
Nice work, Robert!
And by default I think we can have that Panel Closed, and have the one with **Form Options open.**

Soare-Robert-Daniel commented:
Thanks for the feedback.

> And by default I think we can have the fields Panel closed,

This will be done when the PR is ready. For now, it will be opened for easy access during development.

Soare-Robert-Daniel commented on src/blocks/blocks/form/style.scss:
```diff
@@ -62,6 +108,30 @@
 				}
 			}
 
+			&.full-tablet {
+				@media ( max-width: 1024px ) {
+					flex-direction: column;
+					align-items: center;
+
+					.wp-block-button__link {
+						width: 100%;
+						max-width: unset;
+					}
+				}
+			}
+
+			&.full-mobile {
+				@media ( max-width:640px ) {
+					flex-direction: column;
+					align-items: center;
+
+					.wp-block-button__link {
+						width: 100%;
+						max-width: unset;
+					}
+				}
+			}
+
```
@HardeepAsrani, I need to use classes for the Full style on submit button alignment. The problem is that I do not know how to integrate this with the Global CSS Default.

HardeepAsrani commented:
@Soare-Robert-Daniel I think it looks pretty nice, I only have few feedback here: https://www.loom.com/share/25af3348e8c64091a04952deb3b6ceb6 - apart from that, it looks good to go, you can get the design team's go ahead and then we can put it for QA.

Soare-Robert-Daniel commented:
@mghenciu @JohnPixle, you can review the block design.

One question remains, do we want the submit button always to inherit the theme colors? By setting it to inherit, we can no longer use the advantages of global defaults because we can not use CSS vars.

mghenciu commented:
Nice work overall, Robert!
Just 2 design mensions.

- [ ] I think it may be better to call inner child **Field Settings**, instead of just Settings. Emphasising this way that those settings refer to the Field and not the form.
<img width="886" alt="Screenshot 2023-01-04 at 18 46 49" src="https://user-images.githubusercontent.com/52494172/210606506-55661be4-49d0-404f-bc7b-5f7c7996bf6f.png">

- [ ] and the other thing is regarding the **Order** of the fields, in the Style Tab. This is on me, so I also adjusted it in [Figma](https://www.figma.com/file/LRrhO1ovZnXFibXVsezR8a/Otter?node-id=2217%3A28046&t=XURVAgFSvJfasHOi-1), in this order: Form Color, Button, Labels, Input Fields, Border, Helper Text, Submit Messages.

_Essentially the idea here being that the order should be as it's in a normal flow, meaning that Success/Error message are at the bottom. The only exception we did is the button style, which may be used more often, compared to the Submit Messages panel which I assume will not be used a lot._

___
About the Button Color inheritance, so what you are saying is that if we stick to this approach - we can't sync the color in the Global Defaults, right?

The thing we are trying to solve with the Inheritance is when Otter is used with other themes that Neve, and in Patterns. So for example let's say I have a site with Pink buttons, and I import an Otter Pattern that contains a form. As a user, **I would expect the form button to be pink (but of course I can change that).**

Soare-Robert-Daniel commented:
@irinelenache The issues should be fixed now 👌 Also I updated your testing instance.

> * [ ]  The color control for the submit message closes when i click on a color https://vertis.d.pr/v/0926YO

I could not reproduce this. The control should work normal.

Soare-Robert-Daniel commented:
> > Selecting the global default for the button hover text color isn't working https://vertis.d.pr/v/bCiCqM

Fixed. It was a typo in the name.

> > A new added block inherits the global defaults as it should, but if i click Apply all for syncing with global defaults, the style will be not applied https://vertis.d.pr/v/876B62

This is working. I tested in your instance. For a correct update on Global, you need to press Save **and** Update/Save draft (for Post/Page) to see the result.

P.S: Updated your instance with the new build 🙏

irinelenache commented:
@Soare-Robert-Daniel Tested again and all issues are fixed now, thank you 🚀
